swift_scality_backend/diskfile.py

```diff
@@ -138,16 +134,17 @@ def can_zero_copy_send(self):
 
     @utils.trace
     def zero_copy_send(self, wsockfd):
-        (ipaddr, port) = self._filesystem.sproxyd_hosts.next()
+        object_url = self._filesystem.get_url_for_object(self._name)
+        sproxyd = urlparse.urlparse(object_url)
         conn = None
 
         with swift.common.exceptions.ConnectionTimeout(
-                self._filesystem.conn_timeout):
+                self._filesystem._conn_timeout):
```

Comment:
It makes sense to retrieve the value from the `SproxydClient` instance, but not by accessing a protected attribute :wink: